agent/wsclient/client.go

```diff
@@ -122,32 +119,44 @@ type ClientServerImpl struct {
 // 'MakeRequest' can be made after calling this, but responss will not be
 // receivable until 'Serve' is also called.
 func (cs *ClientServerImpl) Connect() error {
+
 	parsedURL, err := url.Parse(cs.URL)
+
 	if err != nil {
 		return err
 	}
 
+	// gorilla/websocket expects the websocket scheme (ws[s]://)
+	parsedURL.Scheme = "wss"
+
 	// NewRequest never returns an error if the url parses and we just verified
 	// it did above
-	request, _ := http.NewRequest("GET", cs.URL, nil)
+	request, _ := http.NewRequest("GET", parsedURL.String(), nil)
+
 	// Sign the request; we'll send its headers via the websocket client which includes the signature
 	utils.SignHTTPRequest(request, cs.Region, ServiceName, cs.CredentialProvider, nil)
 
-	wsConn, err := cs.websocketConn(parsedURL, request)
-	if err != nil {
-		return err
+	timeoutDialer := &net.Dialer{Timeout: wsConnectTimeout}
+	tlsConfig := &tls.Config{ServerName: parsedURL.Host, InsecureSkipVerify: cs.AcceptInvalidCert}
+
+	dialer := websocket.Dialer{
+		ReadBufferSize:  readBufSize,
+		WriteBufferSize: writeBufSize,
+		TLSClientConfig: tlsConfig,
+		Proxy:           http.ProxyFromEnvironment,
```

Comment:
It might be worth it to override the `NO_PROXY` environment variable, since we recommend that it's set very specifically in [our documentation](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/http_proxy_config.html).  Probably something like `169.254.169.254,169.254.170.2,$DOCKER_HOST` (and ensure that `DOCKER_HOST` defaults to the platform default (`/var/run/docker.sock` on Linux, `\\.\pipe\docker_engine` on Windows) when unset.  Another alternative (that would easily allow overriding) is to only set `NO_PROXY` when it's unset in the environment otherwise.